bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/relay/RelayHandler.java

```diff
@@ -176,22 +181,39 @@ private void addChannels(List<String> channelsToAdd, ThingBuilder thingBuilder)
     private Channel createChannel(String channelId) {
         ChannelUID channelUID = new ChannelUID(getThing().getUID(), channelId);
         ChannelTypeUID channelTypeUID = getChannelTypeUID(channelId);
-        return ChannelBuilder.create(channelUID).withType(channelTypeUID).build();
+        @Nullable
+        String itemType = getItemType(channelId);
+        return ChannelBuilder.create(channelUID, itemType).withType(channelTypeUID).build();
```

Comment:
After @mike-bike reported that the channels cannot be linked after having been re-created dynamically, I now try to pass the item type as well. Let's hope this works better. If it doesn't, we might need to pass even more metadata, but I actually wanted to avoid duplicating all the metadata from `thing-types.xml` in the code...

Or we go back to the variant where we just remove channels.

Reply:
Hi David, give it a try. I am back on Monday to test. 
If you decide removing the channels, remember to unlink items before avoiding orphaned links. 

Best regards
Michael

Reply:
JAR is now updated :+1: Thanks for your continuing testing support ❤️